hector_math/include/hector_math/containers/ring_buffer.h

```diff
@@ -0,0 +1,183 @@
+//
+// Created by aljoscha on 02.02.2023.
+//
+
+#ifndef HECTOR_MATH_RING_BUFFER_H
+#define HECTOR_MATH_RING_BUFFER_H
+
+#include <array>
+#include <assert.h>
+#include <type_traits>
+
+namespace hector_math
+{
+
+template<typename T, int MaxSize>
+class RingBuffer
+{
+public:
+  // template parameter controls whether the iterator returns const references
+  template<bool C>
+  struct ring_iterator {
+    using iterator_category = std::forward_iterator_tag;
+    using difference_type = std::ptrdiff_t;
+    using value_type = T;
+    using pointer = T *;
+    using reference = T &;
+    using const_reference = T const &;
+
+    ring_iterator( std::array<T, MaxSize> *items, size_t index, size_t size ) : items_( items ),
+                                                                                index_( index ),
+                                                                                    size_(size)
+    {
+    }
+
+    template<bool Z = C, typename std::enable_if<( !Z ), int>::type * = nullptr>
+    reference operator*()
+    {
+      return ( *items_ )[index_];
+    }
+    template<bool Z = C, typename std::enable_if<( Z ), int>::type * = nullptr>
+    const_reference operator*() const
+    {
+      return ( *items_ )[index_];
+    }
+    template<bool Z = C, typename std::enable_if<( !Z ), int>::type * = nullptr>
+    reference operator->() noexcept
+    {
+      return &( ( *items_ )[index_] );
+    }
+    template<bool Z = C, typename std::enable_if<( Z ), int>::type * = nullptr>
+    const_reference operator->() noexcept
+    {
+      return &( ( *items_ )[index_] );
+    }
```

Comment:
why are some of these noexcept and others not?
I don't think you need to enable_if the const functions since they can and should be available for both the const and non-const iterator.
Btw the `const_reference operator->()` should probably also be const.